### PR TITLE
Tests: Add type-unrestricted version of field mapper getter to SearchContext

### PR DESCRIPTION
**Pull request by rjernst**
This fixes an NPE when using TestSearchContext in SignificanceHeuristicTests.

### CODE REVIEW

File: src/main/java/org/elasticsearch/search/internal/SearchContext.java [hunk 1]
```diff
@@ -349,6 +349,8 @@ public void clearReleasables(Lifetime lifetime) {
 
     public abstract FieldMapper smartNameFieldMapper(String name);
 
+    public abstract FieldMapper smartNameFieldMapperFromAnyType(String name);
```
Comment (dakrone) [inline src/main/java/org/elasticsearch/search/internal/SearchContext.java]: Can you add javadocs about how this is different than `smartNameFieldMapper`? I understand it tries to get it from the `mapperService` usually, but for instance, the percolator context throws an `UnsupportedOperationException` in the regular smartNameFieldMapper method, so I am not sure how they differ.

File: src/test/java/org/elasticsearch/test/TestSearchContext.java [hunk 2]
```diff
@@ -571,6 +571,14 @@ public FieldMappers smartNameFieldMappers(String name) {
...
+    public FieldMapper<?> smartNameFieldMapperFromAnyType(String name) {
+        if (mapperService() != null) {
+            return mapperService().smartNameFieldMapper(name);
+        }
+        return null;
+    }
+
+    @Override
     public MapperService.SmartNameObjectMapper smartNameObjectMapper(String name) {
         return mapperService().smartNameObjectMapper(name, types);
```
Comment (dakrone) [inline src/test/java/org/elasticsearch/test/TestSearchContext.java]: We may want to add the null check here, for consistency?

**Comment by rjernst** [comment]
@dakrone Ok, I think I addressed both issues.

**Comment by dakrone** [comment]
LGTM, too bad we can't just use `Optional` :)

**Pull request merged**